### PR TITLE
added insightface face recognition models

### PR DESCRIPTION
**Pull request by elenash**

### CODE REVIEW

**Comment by kchechil** [comment]
@elenash @snosov1 is it actual? has it gone through full process with accuracy validation etc?
cc @eizamaliev

**Comment by snosov1** [comment]
Yes, we still want to add this model.

There wasn't any work on the integration side (i.e. accuracy checker) - we tested it only manually. Can we do all that's required through Github for this model?

**Comment by kchechil** [comment]
Jenkins please retry a build.

**Comment by kchechil** [comment]
@snosov1 you need to resolve conflicts, confirm with @yssemaev on legal part + include to public OMZ, and run a pre-commit build at least.
But I'd vote for adding support for those models to AC as well. cc @vladimir-dudnik @eaidova

**Comment by eaidova** [comment]
@kchechil ,  Models tested manually using Accuracy Checker, we have configuration files for them in private part of the tool, so Accuracy Checker has the support of these models. If it is necessary, I can move AC configs for these models on github

**Comment by snosov1** [comment]
>  If it is necessary, I can move AC configs for these models on github

I'd vouch for it. Let's try and make this model a "template" contribution, so that when other people want to contribute - we can refer to this one as a demonstrative example.

**Comment by elenash** [comment]
Jenkins please retry a build.

**Comment by elenash** [comment]
@kchechil @snosov1 I've rebased the commit, added accuracy checker configs. I was able to download them via downloader and to convert them to MO in FP32 format. These models are running in the benchmark. Actually, i've done accuracy validation previously (probably in april, that's why cvs files with results  has been already deleted). Is there already any automatic way to run accuracy validation on public models? 
Who is responsible for reviewing? 
@yssemaev Are these models are added to the roadmap and discussed on OS PDT? Anything else is needed from the Legal side?

**Comment by eaidova** [comment]
Jenkins please retry a build

**Comment by yssemaev** [comment]
Hi, these public models should meet release criteria including approvals, when done I'll add a comment/review to PR

**Comment by IRDonch** [comment]
Jenkins: test this please

**Comment by IRDonch** [comment]
@elenash please update `tools/downloader/license.txt`.

**Comment by yssemaev** [comment]
Hi,
Update: while redistributing public models, names of models should be the same like originally published in Open Source and the original license and Copyright terms should be followed.

File: models/public/face-recognition-resnet100-arcface/model.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# limitations under the License.
+
+description: >-
+  Deep face recognition net with ResNet100 backbone and Arcface loss <https://arxiv.org/abs/1801.07698>
+task_type: face_recognition
+files:
+  - name: model-r100-arcface-ms1m-refine-v2.zip
+    sha256: eab0e5bcca81c070f56ffb11a1c32d7b2e574a8bb94f32590e7b48b355c13d1f
+    source: https://www.dropbox.com/s/tj96fsm6t6rq8ye/model-r100-arcface-ms1m-refine-v2.zip?dl=1
+    output: "face_recognition/insightface/mxnet/resnet100/"
```
Comment (IRDonch) [inline models/public/face-recognition-resnet100-arcface/model.yml]: The `output` key is no longer supported (and it was never supported in _this_ position).

File: tools/accuracy_checker/configs/face-reidentification-retail-0095.yml [hunk 2]
```diff
@@ -28,7 +28,7 @@ models:
       - name: lfw
         data_source: LFW/lfw
         annotation_conversion:
-          converter: face_reid_pairwise
+          converter: lfw
```
Comment (IRDonch) [inline tools/accuracy_checker/configs/face-reidentification-retail-0095.yml]: How is this relevant to the rest of this PR?

File: models/public/face-recognition-mobilefacenet-arcface/face-recognition-mobilefacenet-arcface.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+## Example
+
+## Specification
+
+| Metric            | Value         |
+|-------------------|---------------|
+| Type              | Face recognition |
+| GFLOPs            | 0.449         |
+| MParams           | 0.993         |
+| Source framework  | mxnet\*       |
```
Comment (IRDonch) [inline models/public/face-recognition-mobilefacenet-arcface/face-recognition-mobilefacenet-arcface.md]: MXNet

File: tools/accuracy_checker/configs/face-reidentification-retail-0095.yml (hunk 2 above)
Comment (elenash) [reply tools/accuracy_checker/configs/face-reidentification-retail-0095.yml]: @eaidova is it correct?

Comment (eaidova) [reply tools/accuracy_checker/configs/face-reidentification-retail-0095.yml]: I'm sorry, I was going to update it in separated PR. These changes do not have a direct relation with adding new models.

File: models/public/face-recognition-mobilefacenet-arcface/face-recognition-mobilefacenet-arcface.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+### Original model
+
+Image, name - `data`,  shape - `1,3,112,112`, format is `B,C,H,W` where:
+
+- `B` - batch size
+- `C` - channel
+- `H` - height
+- `W` - width
+
+Channel order is `BGR`.
```
Comment (eizamaliev) [inline models/public/face-recognition-mobilefacenet-arcface/face-recognition-mobilefacenet-arcface.md]: By default MXnet uses images in `RGB` channel order. Please, prove me wrong.

File: tools/accuracy_checker/configs/face-reidentification-retail-0095.yml (hunk 2 above)
Comment (elenash) [reply tools/accuracy_checker/configs/face-reidentification-retail-0095.yml]: @IRDonch Should I revert this change by now?

File: models/public/face-recognition-mobilefacenet-arcface/face-recognition-mobilefacenet-arcface.md (hunk 4 above)
Comment (elenash) [reply models/public/face-recognition-mobilefacenet-arcface/face-recognition-mobilefacenet-arcface.md]: Correct, thanks

File: tools/accuracy_checker/configs/face-reidentification-retail-0095.yml (hunk 2 above)
Comment (elenash) [reply tools/accuracy_checker/configs/face-reidentification-retail-0095.yml]: @eaidova I reverted this change in this PR

**Comment by elenash** [comment]
@yssemaev Yuri, I added some refs to original names in markdown files and license file according to Jessica’s guidance

File: models/public/face-recognition-resnet34-arcface/model.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+description: >-
+  Deep face recognition net with ResNet34 backbone and Arcface loss <https://arxiv.org/abs/1801.07698>
+task_type: face_recognition
+files:
+  - name: model-r34-arcface-ms1m-refine-v1.zip
```
Comment (IRDonch) [inline models/public/face-recognition-resnet34-arcface/model.yml]: Please add sizes for all files (they're required since a recent downloader change).
